ehrql/backends/tpp.py

```diff
@@ -296,6 +296,32 @@ def modify_query_variables(self, variables):
         """
     )
 
+    covid_therapeutics = QueryTable(
+        """
+        SELECT DISTINCT
+            Patient_ID AS patient_id,
+            COVID_indication AS covid_indication,
+            Count AS count,
+            CurrentStatus AS current_status,
+            Diagnosis AS diagnosis,
+            FormName AS form_name,
+            Intervention AS intervention,
+            CASIM05_date_of_symptom_onset,
+            MOL1_onset_of_symptoms,
+            SOT02_onset_of_symptoms,
```

Comment:
Hmm...I think we do want to do the select distinct over all the fields so we're only discarding completely duplicate rows, but we don't need to include all these other fields that we don't use in the queryable table. Or does it make no different, efficiency-wise?

We could check the duplicates in the actual data and see if there's a smaller subset of fields that we need in order to remove duplicates (there were only ~35; possibly just the fields we do need to expose are enough)

Reply:
do you mean doing something like "select fields_we_want from (select distinct * from Therapeutics)"? I wouldn't guess that this would be very much more efficient than just "(select * from Therapeutics)" as the db server probably has to load & check the same amount of data, although you can never know with databases unless you actually try it for real.

Alternatively, it would seem weird to apply distinct to more columns that we're returning in a select, as you could still get duplicates in your output (if they differ only by columns that are not returned by the select).

If it's possible to get more knowledge of the source data, though, and we don't need these columns to eliminate the duplicates in this particular dataset then let me know & I'll remove them.

Reply:
Yes, I think that's what I meant. Also, I think in the cohort-extractor version we were trying to be careful to only deduplicate entirely duplicate rows. Maybe it is OK to deduplicate rows that are exact duplicates in the columns of interest; if received date, treatment date, intervention and risk group are all identical, it's probably safe to assume the row is a duplicate. It would inflate `.count_for_patient()` results. Probably also a question for Millie though.

Reply:
Thinking a bit more - as we're planning to have the intervention column in the output, we should only need the aggregated risk_cohort column to deduplicate (i.e. remove CASIM05_date_of_symptom_onset, MOL1_onset_of_symptoms, SOT02_onset_of_symptoms)?

Reply:
i.e. if we had a patient that had two interventions on the same day - so received date, treatment date, and risk group are the same but intervention differs & the *_risk_cohorts differ but follow the value of intervention.

Reply:
removed